mod-source-record-manager-server/src/main/java/org/folio/verticle/consumers/util/MarcImportEventsHandler.java

```diff
@@ -100,11 +107,23 @@ public void handle(JournalService journalService, DataImportEventPayload eventPa
       JournalRecord journalRecord = JournalUtil.buildJournalRecordByEvent(eventPayload,
         journalParams.journalActionType, journalParams.journalEntityType, journalParams.journalActionStatus);
 
-      populateRecordTitleIfNeeded(journalRecord, eventPayload)
-        .onComplete(ar -> journalService.save(JsonObject.mapFrom(journalRecord), tenantId));
+      if (Objects.equals(journalRecord.getEntityType(), PO_LINE)) {
+        processJournalRecordForOrder(journalService, tenantId, journalRecord);
+      } else {
+        populateRecordTitleIfNeeded(journalRecord, eventPayload)
+          .onComplete(ar -> journalService.save(JsonObject.mapFrom(journalRecord), tenantId));
```

Comment:
nitpick: `onComplete(ar -> journalService.save(JsonObject.mapFrom(journalRecord), tenantId))` is repeated here and twice in the `processJournalRecordForOrder` method. I think having  one representation for all code paths would be good.